images/linux/scripts/SoftwareReport/SoftwareReport.Java.psm1

```diff
@@ -1,19 +1,23 @@
 function Get-JavaVersions {
-    $toolcachePath = Join-Path $env:AGENT_TOOLSDIRECTORY "Java_Adopt_jdk"
-    $javaToolcacheVersions = Get-ChildItem $toolcachePath -Name | Sort-Object { [int]$_.Split(".")[0] }
+    $javaToolcacheVersions = Get-ChildItem $env:AGENT_TOOLSDIRECTORY -Depth 1 -Directory |
+        Where-Object {$_.FullName -like "*Java*" -and $_.Name -notlike "Java*"} |
+        Sort-Object { [int]$_.Name.Split(".")[0] }
 
-    return $javaToolcacheVersions | ForEach-Object {
-        $majorVersion = $_.split(".")[0]
-        $fullVersion = $_.Replace("-", "+")
+    $javaToolcacheVersions | ForEach-Object {
```

Comment:
Remove `[array]$existingVersions +=` and replace to `$existingVersions = $javaToolcacheVersions | ForEach-Object {`